File: examples/lh-server/python/04-conditionals/workflow.py
from littlehorse.workflow import Workflow, WorkflowThread


def get_workflow() -> Workflow:
    def wfLogic(wf: WorkflowThread) -> None:
        user_id = wf.declare_str("user-id").required()
        message = wf.declare_str("message").required()
        preferred_contact = wf.declare_str("contact-method")
        preferred_contact.assign(wf.execute("fetch-contact-method", user_id))
        wf.do_if(
            preferred_contact.is_equal_to("COMLINK"),
            if_body=lambda ifHandler: ifHandler.execute(
                "send-comlink-message", user_id, message
            ),
            else_body=lambda elseHandler: elseHandler.execute(
                "send-hologram", user_id, message
            ),
        )

    return Workflow("send-message", wfLogic)

File: examples/lh-server/go/04-conditionals/workflow.go
package main

import "github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"

func Wflogic(wf *littlehorse.WorkflowThread) {
	userID := wf.DeclareStr("user-id").Required()
	message := wf.DeclareStr("message").Required()
	preferredContact := wf.DeclareStr("contact-method")

	preferredContact.Assign(wf.Execute("fetch-contact-method", userID))
	wf.DoIfElse(
		preferredContact.IsEqualTo("COMLINK"),
		func(ifHandler *littlehorse.WorkflowThread) {
			ifHandler.Execute("send-comlink-message", userID, message)
		},
		func(elseHandler *littlehorse.WorkflowThread) {
			elseHandler.Execute("send-hologram", userID, message)
		},
	)
}
